### DISCUSSION TITLE
add optional uid parameter to createUserWithEmailAndPassword

### DISCUSSION THREAD

**Discussion by ShaneZhengNZ** [invertase/react-native-firebase#6471] [Ideas] [answered]
just wondering whether it is possible to add an optional uid parameter to createUserWithEmailAndPassword method?

Is that a limitation of this package or Firebase?

**Comment by mikehardy** [comment, marked as answer]
Hey there! Sorry for the delay, I'm still catching up from a bunch of summer travel.
We stick with the firebase-js-sdk APIs where ever they exist, and we implement what is possible with the firebase-android-sdk and firebase-ios-sdk APIs that we use at the native level

So in this case we have this API which does not allow for an optional uid parameter: https://firebase.google.com/docs/reference/js/v8/firebase.auth.Auth#createuserwithemailandpassword

and at the native level:
- android doesn't have it: https://firebase.google.com/docs/reference/android/com/google/firebase/auth/FirebaseAuth#public-taskauthresult-createuserwithemailandpassword-string-email,-string-password
- ios doesn't have it: https://firebase.google.com/docs/reference/ios/firebaseauth/api/reference/Classes/FIRAuth#-createuserwithemail:password:completion:

So it's not possible
